src/bun.js/api/bun/udp_socket.zig

```diff
@@ -0,0 +1,714 @@
+        globalThis.throw("Cannot construct UDPSocket", .{});
+        return null;
+    }
+
+    pub fn hasPendingActivity(this: *This) callconv(.C) bool {
+        return this.js_refcount > 0;
+    }
+
+    pub fn udpSocket(globalThis: *JSGlobalObject, options: JSValue) JSValue {
+        log("udpSocket", .{});
+
+        const config = UDPSocketConfig.fromJS(globalThis, options) orelse {
+            return .zero;
+        };
+
+        var vm = globalThis.bunVM();
+        var this: *This = vm.allocator.create(This) catch bun.outOfMemory();
+        this.* = This{
+            .socket = undefined,
+            .config = config,
+            .globalThis = globalThis,
+            .send_buf = uws.udp.PacketBuffer.create(),
+            .loop = uws.Loop.get(),
+            .vm = vm,
+        };
+
+        if (uws.udp.Socket.create(
+            this.loop,
+            onData,
+            onDrain,
+            config.hostname,
+            config.port,
+            this,
+        )) |socket| {
+            this.socket = socket;
+        } else {
+            globalThis.throw("Failed to bind socket", .{});
+            return .zero;
+        }
+
+        if (config.connect) |connect| {
+            if (this.socket.connect(connect.address, connect.port) == -1) {
+                globalThis.throw("Failed to connect socket", .{});
+                return .zero;
+            }
+            this.connect_info = .{ .port = connect.port };
+        }
+
+        this.poll_ref.ref(vm);
+        this.config.protect();
+        const thisValue = this.toJS(globalThis);
+        thisValue.ensureStillAlive();
+        this.thisValue = thisValue;
+        return JSC.JSPromise.resolvedPromiseValue(globalThis, thisValue);
+    }
+
+    pub fn callErrorHandler(
+        this: *This,
+        thisValue: JSValue,
+        err: []const JSValue,
+    ) bool {
+        const callback = this.config.on_error;
+        const globalThis = this.globalThis;
+        const vm = globalThis.bunVM();
+
+        if (callback == .zero) {
+            if (err.len > 0)
+                vm.onUnhandledError(globalThis, err[0]);
+
+            return false;
+        }
+
+        const result = callback.callWithThis(globalThis, thisValue, err);
+        if (result.isAnyError()) {
+            vm.onUnhandledError(globalThis, result);
+        }
+
+        return true;
+    }
+
+    pub fn sendMany(this: *This, globalThis: *JSGlobalObject, callframe: *CallFrame) callconv(.C) JSValue {
+        const arguments = callframe.arguments(1);
+        if (arguments.len != 1) {
+            globalThis.throwInvalidArguments("Expected 1 argument, got {}", .{arguments.len});
+            return .zero;
+        }
+
+        const arg0 = arguments.ptr[0];
+
+        var iter = arg0.arrayIterator(globalThis);
+        var i: u16 = 0;
+        var vals: [3]JSValue = .{ .zero, .zero, .zero };
+        while (iter.next()) |val| : (i += 1) {
```

Comment:
shouldn't we check if the length of the array is divisible by 3 and then if it's not throw an error? 

then, we can simplify this code into either sendManyOneAddress or sendMany, and get the next 3 vals, fill the array, etc